ql/src/java/org/apache/hadoop/hive/ql/parse/GenTezUtils.java

```diff
@@ -196,6 +196,11 @@ public MapWork createMapWork(GenTezProcContext context, Operator<?> root,
       mapWork.setIncludedBuckets(ts.getConf().getIncludedBuckets());
     }
 
+    if (ts.getProbeDecodeContext() != null) {
+      // TODO: some operators like VectorPTFEvaluator do not allow the use of Selected take this into account here?
```

Comment:
I think this may be taken into account when we are pushing the SJ predicates down since it would not be valid in that case either (which I believe your logic to create the context is relying on?). Could you verify that?

Reply:
Just to clarify, I am referring to this Precondition check that is used by all PTF operators.
As this optimization enables TS filtering by setting selectedInUse down the pipeline it could make these PreConditions to fail.

We could take this into account when adding valid MJ-TS ops indeed, however it could be still beneficial for other operators on the pipeline that do use the selected array. 
So the real question: is the condition really needed on those operators? 
It does not look to me that it could cause any issues if it is enabled (even if PTF ops dont use it)

https://github.com/apache/hive/blob/cc38bcc5a993304898ba37b8496f13a15d62bf16/ql/src/java/org/apache/hadoop/hive/ql/exec/vector/ptf/VectorPTFEvaluatorDecimalAvg.java#L64